packages/jest-resolve/src/defaultResolver.ts

```diff
@@ -149,3 +155,31 @@ function realpathSync(file: Config.Path): Config.Path {
 function readPackageSync(_: unknown, file: Config.Path): PkgJson {
   return readPackageCached(file);
 }
+
+function createPackageFilter(conditions?: Array<string>) {
+  const options: ResolveExportsOptions = conditions
+    ? {
+        // `resolve.exports` adds `node` unless `browser` is `true`, so let's add this ugly thing
```

Comment:
this (and the one below), is why I'm hesitant in adding this - I don't like how it adds node's defaults (except for a `browser` toggle which turns it off).

I added a comment [here](https://github.com/yarnpkg/berry/pull/2431#discussion_r720221992), but I should probably just make an issue instead